File: cpp/FrogRiverOne.cpp
#include <vector>
#include <set>

using namespace std;

// int solution(int X, vector<int> &A) {

//     set<int>  positions;
//     for(unsigned i = 0; i<A.size(); i++)
//     {
//         positions.insert(A[i]);
//         if(positions.size()==unsigned(X)) return i;
//     }
//     return -1;
// }

int solution(int X, vector<int> &A) {
    vector<int> vec(X, 0);
    int cnt = 0;
    for(unsigned i = 0; i<A.size(); i++)
    {
        if(A[i]<=X && A[i]>0)
        {
            vec[A[i]-1]++;
            if(vec[A[i]-1] == 1) 
            {
                cnt++;
                if (cnt==X) return int(i);
            }
        }
    }
    return -1;
}

File: python/FrogRiverOne.py
def solution(X, A):
    x = [0]*X

    counter = 0
    i = 0

    for i in range(0, len(A)):
        if A[i]>0 and A[i]<=X:
            x[A[i]-1]+=1
            if x[A[i]-1]==1:
                counter+=1
                if counter == X:
                    return i

    return -1
